src/transformers/modeling_tf_utils.py

```diff
@@ -247,12 +247,83 @@ def compute_loss(self, labels, logits):
         return loss_fn(next_sentence_label, next_sentence_reduced_logits)
 
 
-def input_processing(func, input_ids, **kwargs):
+def booleans_processing(config, **kwargs):
+    """
+    Process the input booleans of each model in order to be sure they are compliant with the execution mode (eager or
+    graph)
+
+    Args:
+        config (:class:`~transformers.PretrainedConfig`):
+            The config of the running model.
+        **kwargs:
+            The boolean parameters
+
+    Returns:
+        A dictionary with the proper values for each boolean
+    """
+    final_booleans = {}
+
+    if tf.executing_eagerly():
+        final_booleans["output_attentions"] = (
+            kwargs["output_attentions"] if kwargs["output_attentions"] is not None else config.output_attentions
+        )
+        final_booleans["output_hidden_states"] = (
+            kwargs["output_hidden_states"]
+            if kwargs["output_hidden_states"] is not None
+            else config.output_hidden_states
+        )
+
+        if "return_dict" in kwargs:
+            final_booleans["return_dict"] = (
+                kwargs["return_dict"] if kwargs["return_dict"] is not None else config.return_dict
+            )
+
+        if "use_cache" in kwargs:
+            final_booleans["use_cache"] = kwargs["use_cache"] if kwargs["use_cache"] is not None else config.use_cache
+    else:
+        if (
+            kwargs["output_attentions"] is not None
+            or kwargs["output_hidden_states"] is not None
+            or ("use_cache" in kwargs and kwargs["output_hidden_states"] is not None)
+            or ("return_dict" in kwargs and kwargs["return_dict"] is not None)
+        ):
+            logger.warn(
+                "Cannot update the boolean parameters behavior in graph mode and the return_dict parameter is always True in that mode."
+            )
+
+        final_booleans["output_attentions"] = config.output_attentions
+        final_booleans["output_hidden_states"] = config.output_hidden_states
+
+        if "return_dict" in kwargs:
+            final_booleans["return_dict"] = True
```

Comment:
I thought this is always True -> this looks like if someone sets `config.use_return_dict=False` => `return_dict` would stay False?

Reply:
ah ok never mind! It's always in kwargs because every forward has it...but do we really need the `if "return_dict" in kwags then? -> it should always be in there no?

Reply:
No because only the values in `final_booleans` are taken into account afterwards in the model/layers.